canonical-web-and-design/ubuntu.com
# Renewals UI

# Done

Built the Stripe payment UI

# QA

- Visit /advantage
- Login with SSO
- Get Scott to set you up with test contracts and renewals
- Try as hard as you can to break the payment flow with the [test card numbers](https://stripe.com/docs/testing) - there are a lot of them on that page, some of them are designed to return specific responses, so test as many as you can and verify that any errors are what you'd expect to see (if not, document it and tell Scott).

- Check out this branch locally
- Run `npm run js-tests`, see that none of them fail
- Run the python tests too, see that they don't fail
  - Verify that the test will catch broken layouts by breaking the layout. As a suggestion, introduce a typo to the script tag source on [line 197](https://github.com/canonical-web-and-design/ubuntu.com/pull/7310/files#diff-bd58ce7b6e12af75432ead96c7f84159R196) in the renewal modal template, then run the Python tests again.

Fixes #6203
